### PR TITLE
ci: replace travis with actions

### PR DESCRIPTION
**Pull request by gr2m**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      - "*.x" # maintenance releases branches
+
+jobs:
+  release:
+    name: release
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+      - run: npm ci
```
Comment (travi) [inline .github/workflows/release.yml:17]: what would you think of using https://github.com/bahmutov/npm-install for the install step? automatically implements [caching in an optimized way](https://glebbahmutov.com/blog/do-not-let-npm-cache-snowball/) for us

Comment (gr2m) [reply .github/workflows/release.yml:17]: sure, once we add lock files?

**Comment by gr2m** [comment]
eeehm ... what? https://github.com/semantic-release/github/pull/298/checks?check_run_id=1397936168 some line ending foo with windows?

Let me remove the linting and only run it once

**Comment by travi** [comment]
> some line ending foo with windows?

could be worth getting a `.gitattributes` in place, like [this](https://github.com/form8ion/replace-travis-ci-with-github-actions/blob/master/.gitattributes) to handle the cross-os line ending issues

File: .github/workflows/release.yml (hunk 1 above)
Comment (travi) [reply .github/workflows/release.yml:17]: lockfiles shouldnt have much of an impact specifically for this. internally, it runs `npm ci`, just like you did, but also configures caching. the caching should work regardless of lockfile

**Comment by gr2m** [comment]
I'm pretty sure the problem will be resolve by implementing automated formatting fixes with `prettier` as part of CI.

Shall I add a package-lock.json as part of this PR?

**Comment by travi** [comment]
> Shall I add a package-lock.json as part of this PR?

works for me

**Comment by gr2m** [comment]
this is ready to ship now

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: bahmutov/npm-install@v1
+      - run: npm run test:ci
```
Comment (travi) [inline .github/workflows/test.yml:34]: do we need a todo here to go back to the normal `test` script once we are able to?

**Review by travi — APPROVED** [review]

**Comment by travi** [comment]
one more question. rather than deleting the `.npmrc` completely, what would you think of enabling [engine-strict](https://docs.npmjs.com/cli/v6/using-npm/config#engine-strict) to help us catch breaking changes from our dependencies based on node version?

**Comment by gr2m** [comment]
> one more question. rather than deleting the `.npmrc` completely, what would you think of enabling [engine-strict](https://docs.npmjs.com/cli/v6/using-npm/config#engine-strict) to help us catch breaking changes from our dependencies based on node version?

I wonder if that is something we could enable on CI only, so we don't need the additional dot file?

Let me merge this first so we have tests running again, but let's keep discussing this point here

**Pull request merged**